src/app/plugins/pubsub.plugin.ts

```diff
@@ -1,23 +1,142 @@
+import Constants from '../../common/constants';
 import { Plugin } from '../../common/plugin';
-import { IContainer, IMessage, ChannelType } from '../../common/types';
+import { ChannelType, IContainer, IHttpResponse, IMessage } from '../../common/types';
+import { MessageEmbed } from 'discord.js';
 
 export class PubSubPlugin extends Plugin {
   public name: string = 'Pub Sub Plugin';
-  public description: string = 'Are publix chicken tender subs on sale?';
-  public usage: string = 'pubsub';
-  public pluginAlias = [];
-  public permission: ChannelType = ChannelType.Bot;
+  public description: string = 'Get prices and steamy pictures of the subs you need in your life.';
+  public usage: string = 'pubSub <subName (optional)> | <"list" | "types" (optional)>';
+  public pluginAlias = ['sub', 'subs', 'sandwich', 'samwich', 'sarnie'];
+  public permission: ChannelType = ChannelType.Public;
+  public pluginChannelName: string = Constants.Channels.Public.Food;
+
+  private _API_URL: string = 'https://api.pubsub-api.dev';
+  private _VALID_KEYS: string[] = ['random', 'list', 'types', ''];
+  private _DEFAULT_INPUT: string = 'chicken-tenders';
+  private _SUBS: string[] = [];
+  private _EMBED_LIST = new MessageEmbed();
+
+  private _SUB_UPD_THRESH: number = 1000 * 60 * 60 * 24; // in ms, one day.
+  private _LAST_UPD_TIME: number = 0;
 
   constructor(public container: IContainer) {
     super();
+    this._updateList();
+  }
+
+  private async _updateList() {
+    this.container.httpService
+      .get(`${this._API_URL}/allsubs/`)
+      .then((response: IHttpResponse) => {
+        const subs = response.data;
+
+        this._SUBS = subs.map((subData: { name: string }) => subData.name);
+      })
+      .catch((err) => this.container.loggerService.warn(err));
   }
 
   public async execute(message: IMessage, args?: string[]) {
-    const state = this.container.storeService.get('pubSubStore').state;
-    if (state?.onSale) {
-      message.reply('Chicken tender subs are on sale :eyes:');
-    } else {
-      message.reply('Chicken tender subs are **NOT** on sale.');
+    const input = (args || []).join('-').toLowerCase() || this._DEFAULT_INPUT;
+
+    if (input === 'list' || input === 'types') {
+      //Simply return the list of supported subs
+      await this._generateEmbedList();
+      message.reply(this._EMBED_LIST);
```

Comment:
i prefer that `_generateEmbedList` returns the embed list (but can also cache for lazy) so that way we can

```
message.reply(await this._generateEmbedList());
```

and not care about the details of `_generateEmbedList` other than that it returns the embed list we want.